File: frame/map-frame-project/MapFrameCoreModule/src/main/java/com/grandtech/mapframe/core/editor/IEditor.java
package com.grandtech.mapframe.core.editor;

import android.content.Context;

import com.grandtech.mapframe.core.util.FeatureSet;
import com.mapbox.geojson.Geometry;

import java.util.Map;

/**
 *
 * @author fs
 * @date 2021/4/16
 */

public interface IEditor {

    /**
     * 合并
     * @param flag
     * @param selectSet
     */
    public void unionFeatures(final Context context, final String flag, Boolean updateTargetSource , final IToolInterceptor iToolInterceptor, final Map<String, FeatureSet> selectSet);

    /**
     * 删除
     * @param flag
     * @param selectSet
     */
    public void delFeatures(final String flag, final IToolInterceptor iToolInterceptor, final Map<String, FeatureSet> selectSet);

    /**
     * 分割
     * @param flag
     * @param selectSet
     * @param geometry
     */
    public void splitFeatures(final String flag, Boolean updateTargetSource ,final IToolInterceptor iToolInterceptor, final Map<String, FeatureSet> selectSet, final Geometry geometry);

    /**
     * 扣除
     * @param flag
     * @param selectSet
     * @param geometry
     */
    public void digFeatures(final String flag, Boolean updateTargetSource ,final IToolInterceptor iToolInterceptor, final Map<String, FeatureSet> selectSet, final Geometry geometry);

    /**
     * 新增
     * @param flag
     * @param geometry
     */
    public void drawFeature(final String flag, Boolean updateTargetSource ,final IToolInterceptor iToolInterceptor, final Geometry geometry);

    /**
     * 新增批量
     * @param flag
     * @param selectSet
     */
    public void addFeatures(final String flag, Boolean updateTargetSource ,final IToolInterceptor iToolInterceptor, final Map<String, FeatureSet> selectSet);

    /**
     * 编辑
     * @param flag
     * @param selectSet
     * @param newGeometry
     */
    public void editFeature(final String flag, Boolean updateTargetSource ,final IToolInterceptor iToolInterceptor, final Map<String, FeatureSet> selectSet, final Geometry newGeometry);


}

File: frame/map-frame-project/MapFrameCoreModuleV10/src/main/java/com/gradtech/mapframev10/core/offline/editor/IEditor.kt
package com.gradtech.mapframev10.core.offline.editor

import android.content.Context
import com.gradtech.mapframev10.core.util.FeatureSet
import com.mapbox.geojson.Geometry

/**
 * @ClassName IEditor
 * @Description TODO
 * @Author: fs
 * @Date: 2022/12/20 14:50
 * @Version 2.0
 */
interface IEditor {
    /**
     * 合并
     * @param flag
     * @param selectSet
     */
    fun unionFeatures(context: Context, flag: String, updateTargetSource: Boolean, iToolInterceptor: IToolInterceptor, selectSet: Map<String, FeatureSet>)

    /**
     * 删除
     * @param flag
     * @param selectSet
     */
    fun delFeatures(flag: String, iToolInterceptor: IToolInterceptor, selectSet: Map<String, FeatureSet>)

    /**
     * 分割
     * @param flag
     * @param selectSet
     * @param geometry
     */
    fun splitFeatures(flag: String, updateTargetSource: Boolean, iToolInterceptor: IToolInterceptor, selectSet: Map<String, FeatureSet>, geometry: Geometry)

    /**
     * 扣除
     * @param flag
     * @param selectSet
     * @param geometry
     */
    fun digFeatures(flag: String, updateTargetSource: Boolean, iToolInterceptor: IToolInterceptor, selectSet: Map<String, FeatureSet>, geometry: Geometry)

    /**
     * 新增
     * @param flag
     * @param geometry
     */
    fun drawFeature(flag: String, updateTargetSource: Boolean, iToolInterceptor: IToolInterceptor, geometry: Geometry)

    /**
     * 新增批量
     * @param flag
     * @param selectSet
     */
    fun addFeatures(flag: String, updateTargetSource: Boolean, iToolInterceptor: IToolInterceptor, selectSet: Map<String, FeatureSet>)

    /**
     * 编辑
     * @param flag
     * @param selectSet
     * @param newGeometry
     */
    fun editFeature(flag: String, updateTargetSource: Boolean, iToolInterceptor: IToolInterceptor, selectSet: Map<String, FeatureSet>, newGeometry: Geometry)
}